### ISSUE TITLE
request: Plugin in AU format

### ISSUE THREAD

**Issue by gdreyband** [jpcima/ADLplug#48]
Hey @jpcima I'm so glad I found this! I was an occasional contributor to JUCEOPLVSTi and I'm so glad to see an OPL3 plugin that's available on Mac natively! I'm playing around with the standalone version now but I'm a Logic user and I'd love to see AU plugin formats supported.

Let me know if that's possible.

**Comment by jpcima** [comment]
Hello @gdreyband
This project is based on Juce plugin framework, which means the code should support directly the Audio Unit specification.

It's not yet readily available only for a few practical reasons.
1. I'm not an owner of Mac, meaning it's difficult to test things personally, and I rely on external feedback which I receive.
2. It's been low on the priority list, because now one has requested it before you did.
3. Last time I looked, I did not understand how to introduce this to automatic builds, because the AU way if building has confused me; but I will re-examine this.

As a user of Mac, you have a solution to construct AU which has a possibility to work, if you are adventurous enough to build the source code.
These subdirectories contain Xcode projects, which hopefully are able to build.
- `project/adl/juce/Builds/MacOSX`
- `project/opn/juce/Builds/MacOSX`

I cannot guarantee this method succeeds, because I have never tried it.

**Comment by jpcima** [comment]
@gdreyband I have some AU builds under the [latest](https://github.com/jpcima/ADLplug/releases/tag/latest) dev release.
Are they working?

**Comment by gdreyband** [comment]
I just tested the one under latest, I got an incompatible plug-in error so it doesn't work unfortunately. I can try to do a build of the source over the next few days.

**Comment by jpcima** [comment]
I have made a new iteration of Audio Unit. You're welcome to try this one.
If still not working, please indicate the computer model, OS version, and processor architecture.

- fixed a defect of AU bundle construction, where the resource fork is not at the right location
- the Xcode project files have been updated
- symbolic links have been added in the package (for a more dmg-like experience, but not quite still). I should provide a DS_Store layout
- files remain distributed as 7z; I have thought of changing it to something more conventional, like a dmg or xip, however 7z is compressed twice more compact.

@sub26nico if you have time and opportunity for this, do you mind to try loading these Audio Units ?
It will be very appreciated.

Binaries are found at the usual place.
[ADL](https://github.com/jpcima/ADLplug/releases/download/latest/ADLplug-dev-mac.7z)
[OPN](https://github.com/jpcima/ADLplug/releases/download/latest/OPNplug-dev-mac.7z)

**Comment by gdreyband** [comment]
For reference, I'm on a Mac on OS 10.12.6, using Logic Pro X 10.4.4

It works, although it fails the plug-in check. If I override that I can use it within Logic.

There are some miscellaneous bugs I found which I can file separately:

For both AU and Standalone
- In the bank [AIL] Ultimate Soccer Manager, certain patches just produce a generic white noise sweep. They're 2-OP and they all appear to have the same or very similar parameters. Switching from FM 1-2 to AM 1-2 produces a vacuum-like noise that doesn't decay.
Patches, for reference: 29-31, 44, 50-55, 65-67

**Comment by sub26nico** [comment]
@jpcima , I've quickly tested the binaries, all seems good. Standalone works fine and made sounds. 
Au,Vst and Vst3 works in Reaper, Au,VST and lv2 works in Ardour. I have just put the plugins on a midi track and play with the keyboard on the UIs, no more. I'm on El Capitan (10.11). I have no much times at the moment but feel free to ping me later if you want deeper tests.

**Comment by jpcima** [comment]
>It works, although it fails the plug-in check. If I override that I can use it within Logic.

Yes. The plugin has a known defect ; it is unable to handle adequately parameters and states while it is in a *disabled* situation. It was an oversight of the design, and is most likely why validation is not passing.

With a somewhat hackish workaround used, it seemed to operate fine in practical cases.
An ideal solution takes some work and more rigorous testing.

>In the bank [AIL] Ultimate Soccer Manager, certain patches just produce a generic white noise sweep. They're 2-OP and they all appear to have the same or very similar parameters. Switching from FM 1-2 to AM 1-2 produces a vacuum-like noise that doesn't decay.

It's the original game bank and most likely it never used these instrument slots.
About *noise that doesn't decay*, you are correct, and more annoyingly the panic button is not able to silence these channels.
@Wohlstand this seems like a case where `adl_panic` is failing.

@sub26nico
Thanks also for confirming AU working.
I have managed a vmware setup which gives me access to compiling and AU lab, but none of the big DAWs though.

**Comment by Wohlstand** [comment]
> this seems like a case where adl_panic is failing.

Looks like is need to override instruments by correct zeroing data, and also be careful against rhythm-mode which makes three channels to act differently... Will check this at my home, for now I'm at my friend
